ResearchKit/Printing/ORKPrintFormatter.m

```diff
@@ -0,0 +1,231 @@
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKPrintFormatter.h"
+#import "ORKHTMLPrintingTemplate.h"
+#import "ORKQuestionStep.h"
+#import "ORKQuestionStep_Internal.h"
+#import "ORKFormStep.h"
+#import "ORKFormItem_Internal.h"
+#import "ORKReviewStep.h"
+#import "ORKHelpers.h"
+#import "ORKAnswerFormat_Internal.h"
+#import "ORKResult_Private.h"
+#import "ORKChoiceAnswerFormatHelper.h"
+#import <AVFoundation/AVFoundation.h>
+
+
+static const CGFloat POINTS_PER_INCH = 72;
+
+#pragma mark - ORKHTMLPrintFormatter
+
+@implementation ORKHTMLPrintFormatter
+
+- (instancetype)init {
+    self = [super initWithMarkupText:@""];
+    if (self) {
+        _styleSheetContent = @"";
+        self.perPageContentInsets = UIEdgeInsetsMake(POINTS_PER_INCH * 0.5f, POINTS_PER_INCH * 0.5f, POINTS_PER_INCH * 0.5f, POINTS_PER_INCH * 0.5f);
+    }
+    return self;
+}
+
+- (void)setSteps:(NSArray<ORKStep *> *)steps withResult:(id<ORKTaskResultSource>)result {
+    NSMutableArray *formatableSteps = [[NSMutableArray alloc] initWithArray:steps];
+    for (ORKStep *step in steps) {
+        if (self.delegate && [self.delegate respondsToSelector:@selector(printFormatter:shouldFormatStep:withResult:)]) {
+            if (![self.delegate printFormatter:self shouldFormatStep:step withResult:[result stepResultForStepIdentifier:step.identifier]]) {
+                [formatableSteps removeObject:step];
+            }
+        }
+    }
+    NSString *body = @"";
+    for (ORKStep *step in formatableSteps) {
+        body = [@[body, [self HTMLFromStep:step withResult:[result stepResultForStepIdentifier:step.identifier] addSurroundingHTMLTags:NO]] componentsJoinedByString:@""];
+    }
+    if ([_styleSheetContent isEqualToString:@""]) {
+        _styleSheetContent = [NSString stringWithContentsOfURL:[ORKBundle() URLForResource:@"HTMLPrintingStylesheet" withExtension:@"css"] encoding:NSUTF8StringEncoding error:nil];
+    }
+    [self setMarkupText:[NSString stringWithFormat:[ORKHTMLPrintingTemplate html], _styleSheetContent, body]];
```

Comment:
Should allow subclassing of `ORKHTMLPrintingTemplate`? Maybe a configurable parameter?

```
@property (nonatomic, nullable) ORKHTMLPrintingTemplate *template;
```